submissions/ses3332002/friends-app/app.js

```diff
@@ -0,0 +1,76 @@
+"use strict";
+const hamburgerMenu = document.querySelector(".hamburger_menu");
+const mainArea = document.querySelector("main");
+const optionForm = document.forms.sorting_n_filtering;
+let receivedData = [];
+let dataToRender = [];
+let filterCriterionSex = () => true;
+let filterCriterionAge = () => true;
+let filterCriterionName = () => true;
+let sortCriterion = () => true;
+as_is.checked = true;
```

Comment:
Where are these variables declared and used?

Reply:
I'm addressing DOM elements through their ID's.
Such a hack like using a button in a form without any listener to reload the page.
I use it to clear filters and sorting fields in form, because with common "reload" in browser they still keep old values, despite the HTML attributes.

Reply:
Please don’t use id-named global variables to access elements.
This behavior is described in the specification, so it’s kind of standard. But it is supported mainly for compatibility.

The browser tries to help us by mixing namespaces of JS and DOM. That’s fine for simple scripts, inlined into HTML, but generally isn’t a good thing. There may be naming conflicts. Also, when one reads JS code and doesn’t have HTML in view, it’s not obvious where the variable comes from.

In real life document.getElementById is the preferred method.

Reply:
Ok, I'll remake it.

Reply:
I remade it.